src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -462,10 +463,27 @@ private async Task RefreshKeyValueCollections()
 
                 try
                 {
-                    IEnumerable<ConfigurationSetting> currentKeyValues = _applicationSettings.Values.Where(kv =>
+                    IEnumerable<ConfigurationSetting> currentKeyValues;
+
+                    // Check if the user-provided feature flag filter ends with an unescaped * character:
+                    // (?<!\\)    Matches if the preceding character is not a backslash
+                    // (?:\\\\)*  Matches any number of occurrences of two backslashes
+                    // \*$        Matches if * is the last character
+                    if (Regex.IsMatch(changeWatcher.Key, @"(?<!\\)(?:\\\\)*\*$"))
                     {
-                        return kv.Key.StartsWith(changeWatcher.Key) && kv.Label == changeWatcher.Label.NormalizeNull();
-                    });
+                        // Get current application settings starting with changeWatcher.Key, excluding the last * character
+                        currentKeyValues = _applicationSettings.Values.Where(kv =>
+                        {
+                            return kv.Key.StartsWith(changeWatcher.Key.Substring(0, changeWatcher.Key.Length - 1)) && kv.Label == changeWatcher.Label.NormalizeNull();
+                        });
+                    }

```

Comment:
Escaped `*` is a complex problem to solve. We would need to implement the same search logic and unescape logic as the backend. I'm thinking we may want to throw an argument exception if someone passes a filter that ends in `\*`.

Reply:
We won't support that scenario, but that scenario means somebody would want to select a single feature flag that ends with '*'. Seems rare.

---
In reply to: [597895109](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/234#discussion_r597895109) [](ancestors = 597895109)

Reply:
We officially don't support * in feature flag names, but it is still possible to create a feature flag named MyApp* using our REST API. So if someone wants to register this feature flag, their only option is to provide "MyApp\*" as the key filter.

Reply:
I get that, but it seems so rare I don't think it's worth complicating our client to support it.

Reply:
If someone asks for it we can always add it. But at this moment, no one is even asking for singular select of a feature flag.

Reply:
Consider "MyApp\\*" key filter. In this case, the `\` character is escaped. So the user is requesting all feature flags that begin with "MyApp\". Should this be a valid request?

Reply:
It's a valid request. I don't think we should support it.